tests/integration/test_interface.py

```diff
@@ -17,27 +18,77 @@
 
 
 @pytest.mark.parametrize(
-    "data_dir_name, noising_function",
+    "data_dir_name, noising_function, use_sample_data",
     [
-        ("decennial_census_observer", generate_decennial_census),
-        ("household_survey_observer_acs", generate_american_communities_survey),
-        ("household_survey_observer_cps", generate_current_population_survey),
-        ("social_security_observer", generate_social_security),
-        ("tax_w2_observer", generate_taxes_w2_and_1099),
-        ("wic_observer", generate_women_infants_and_children),
-        ("tax 1040", "todo"),
+        ("decennial_census_observer", generate_decennial_census, True),
+        ("decennial_census_observer", generate_decennial_census, False),
+        ("household_survey_observer_acs", generate_american_communities_survey, True),
+        ("household_survey_observer_acs", generate_american_communities_survey, False),
+        ("household_survey_observer_cps", generate_current_population_survey, True),
+        ("household_survey_observer_cps", generate_current_population_survey, False),
+        ("social_security_observer", generate_social_security, True),
+        ("social_security_observer", generate_social_security, False),
+        ("tax_w2_observer", generate_taxes_w2_and_1099, True),
+        ("tax_w2_observer", generate_taxes_w2_and_1099, False),
+        ("wic_observer", generate_women_infants_and_children, True),
+        ("wic_observer", generate_women_infants_and_children, False),
+        ("tax 1040", "todo", True),
+        ("tax 1040", "todo", False),
     ],
 )
-def test_generate_form(data_dir_name: str, noising_function: Callable):
+def test_generate_form(
```

Comment:
I'm pretty confused with what you're trying to test here. Would it be better/cleaner to have a separate test? It's unclear to me what you're testing.

Reply:
It could be a separate test but it's going to be a copy/paste job so it felt simpler to parameterize. It's testing exactly the same thing as originally but now with the option of a user providing a root data path (with more than one dataset to load) instead of relying on the default sample data.

Reply:
I pulled out the extra nonsense into a helper function to hopefully make the test more readable. Is that better or do you think it should be a separate test?

Reply:
This is great. We should do something similar for the _generate_form (but not necessarily as part of this PR) to modularize the pieces (getting the data, doing transformations of types, etc.) It makes testing easier but also makes reading the code and understanding the separation of concerns much better.